sway-fmt-v2/src/error.rs

```diff
@@ -9,6 +9,8 @@ pub enum FormatterError {
     FormatError(#[from] std::fmt::Error),
     #[error("Error while lexing file: {0}")]
     LexError(#[from] sway_parse::LexError),
+    #[error("Error while adding comments")]
```

Comment:
like maybe include some extra pointer to information as the above errors do

Reply:
Hmm, not so sure if we have some extra information to print here _for now,_ because this error is produced only if there are no spans collected for the given unformatted/formatted code. Which can only happen if the input is empty. But in that case, this error is still not possible because this check is done _while_ traversing the given input. So if there is something that we can traverse and do this check this means there are some `span`s present in the input. So basically I introduced this because I do not like `.unwrap()` stuff around 😄 also I believe this error will need to be re-used in the future when we are refactoring this part of the code for a better efficient approach. Once we have some other cases where we are producing this error, we will have some other information to print just like the above ones.

So this was more like a preferred style thingy over `.unwrap()` (they are interchangeable in this context as this will not fail 😄) but if you prefer having an explicit `anyhow!("..")` kinda approach other than this approach I can also go that way too.

Reply:
Okay great! Just wanted to see where your head was at (: good work!